server/routes/dashBoard.js

```diff
@@ -0,0 +1,251 @@
+const express = require("express");
+const router = express.Router();
+const { updateData, decodeToken } = require("../util/util");
+
+// Models;
+const { Task, Column, Dashboard } = require("../models/Dashboard");
+
+//@CreateBoard
+router.post("/getDashboard", async (req, res) => {
+  const { token } = req.body;
+  let userId = await decodeToken(token);
+
+  try {
+    let dashboard = await Dashboard.findOne({ user: userId });
+    res.send(dashboard);
+  } catch (err) {
+    res.status(404).json({ error: "Dashboard does not exist" });
+  }
+});
+
+//Add Dashboard @in progress need to update UserId
+router.post("/addDashBoard", async (req, res) => {
+  const { dashboardTitle, token } = req.body;
+
+  let userId = await decodeToken(token);
+
+  if (!userId) {
+    return res.status(401).json({ error: "Token is not valid" });
+  }
+
+  try {
+    // Initial states
+    const task1 = new Task({
+      content: "This is Your Task!",
+      description: "",
+      deadline: "",
+      comments: [],
+      tag: {},
+      action: {}
+    });
+    const column1 = new Column({
+      title: "First column",
+      taskOrder: [task1.id],
+      tasks: { [task1._id]: task1 }
+    });
+    const column2 = new Column({
+      title: "Second column",
+      taskOrder: [],
+      tasks: {}
+    });
+
+    const newDashBoard = new Dashboard({
+      user: userId,
+      dashboardTitle: dashboardTitle,
+      columns: { [column1._id]: column1, [column2._id]: column2 },
+      columnOrder: [column1.id, column2.id]
+    });
+
+    let data = await newDashBoard.save();
+    res.send(data);
+  } catch (err) {
+    res.status(500).json({ error: "Failed to add dashboard" });
+  }
+});
+
+// Add a column @Done
+router.post("/addColumn", async (req, res) => {
+  const { dashBoardId, columnTitle } = req.body;
+  try {
+    const newColumn = new Column({
+      columnTitle,
+      tasks: [],
+      taskOrder: []
+    });
+
+    //
+    let newColumns = {};
+    let Board = await Dashboard.findOne({ _id: dashBoardId });
+
+    for (const key of Board.columns.keys()) {
+      let item = Board.columns.get(key);
+      newColumns[item.id] = item;
+    }
+
+    newColumns = {
+      ...newColumns,
+      [newColumn._id]: newColumn
+    };
+
```

Comment:
It doens't work if instead you do this?
```
board.columns.set(newColum._id,newColumn)
board.columnOrder.push(newColumn._id)
board.save()
```

Reply:
Thanks for reviewing.
I tried your code, but for some reasons, it returns error.
my original code works with postman and request from FE (I tried in other branch).
Could you give me sample more code about it?

This is the code I tried with your code.

```
  const newColumn = new Column({
      columnTitle,
      tasks: [],
      taskOrder: []
    });

    //
    let newColumns = {};
    let board = await Dashboard.findOne({ _id: dashboardId });

    for (const key of board.columns.keys()) {
      let item = board.columns.get(key);
      newColumns[item.id] = item;
    }

    newColumns = {
      ...newColumns,
      [newColumn._id]: newColumn
    };

   board.columns.set(newColum._id, newColumn); <- I also tried with await
   board.columnOrder.push(newColumn._id); <- I also tried with await
   board.save();  <- I also tried with await
```

Reply:
It does work, the mistake was that _id is an object, not a string. So all we had to do is change it to :

`newColum._id.toString()`